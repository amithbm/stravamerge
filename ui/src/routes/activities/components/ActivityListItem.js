import React from 'react'
import {Row,Col} from 'react-bootstrap'
import TitleLabel from '../../common/components/TitleLabel'
import { Link } from 'react-router'

export default ({activity, includeHyperlinks}) => {
    const compareSegmentsLink = '/compareSegments/' + activity.id
    const compareSegments = activity.hasSegments ? <Link className='m-r-2' to={compareSegmentsLink}>Compare Segments</Link> : undefined
    const routeProfileLink = '/routeProfile/' + activity.id
    const routeProfile = activity.hasSegments ? <Link className='m-r-2' to={routeProfileLink}>3D Route Profile</Link> : undefined
    const Checkbox = props => (
        <input type="checkbox" {...props} />
      )
    return <div className='m-b-3'>
        <Row>
            <label>
                <Checkbox
                />
                <span>{activity.startDate.toDateString()} - {Math.floor(activity.distance/100)/10} km - {Math.floor(activity.averageSpeed*10)/10} kmph</span>
                </label>

            {/* <Col xs={10}>{activity.startDate.toDateString()}</Col> */}
                {/* <table className='table-bordered'>
                    <thead>
                        <th width="0"><span className='m-r'><em>Date</em></span></th>
                        <th width="0"><span className='m-r'><em>Distance</em></span></th>
                        <th width="0"><span className='m-r'><em>Avg Speed</em></span></th>
                    </thead>
                    <tbody>
                        <tr>
                            <td width="0"><span className='m-r'>{activity.startDate.toDateString()}</span></td>
                            <td width="0"><span className='m-r'>{Math.floor(activity.distance/100)/10} km</span></td>
                            <td width="0"><span className='m-r'>{Math.floor(activity.averageSpeed*10)/10} kph</span></td>
                        </tr>
                    </tbody>
                </table> */}
            {/* <Col xs={10}>{Math.floor(activity.distance/100)/10} km</Col> */}
            {/* <Col xs={10}>{Math.floor(activity.averageSpeed*10)/10} kmph</Col> */}
        </Row>
    </div>
}