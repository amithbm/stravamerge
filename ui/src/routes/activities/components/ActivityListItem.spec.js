import React from 'react'
import renderer from 'react-test-renderer'
import ActivityListItem from './ActivityListItem'
import { getActivity } from '../../../testing/testData'

describe('components', () => {
    describe('<ActivityListItem />', () => {
        it('renders with hyperlinks', () => {
            const activity = getActivity()
            const component = renderer.create(<ActivityListItem activity={activity} includeHyperlinks={true} />)
            const tree = component.toJSON()
            expect(tree).toMatchSnapshot()
        })

        it('renders without hyperlinks when there are no segments', () => {
            const activity = getActivity()
            activity.hasSegments = false
            const component = renderer.create(<ActivityListItem activity={activity} includeHyperlinks={true} />)
            const tree = component.toJSON()
            expect(tree).toMatchSnapshot()
        })

        it('renders without hyperlinks when there are segments but includeHyperlinks is false', () => {
            const activity = getActivity()
            const component = renderer.create(<ActivityListItem activity={activity} includeHyperlinks={false} />)
            const tree = component.toJSON()
            expect(tree).toMatchSnapshot()
        })
    })
})
