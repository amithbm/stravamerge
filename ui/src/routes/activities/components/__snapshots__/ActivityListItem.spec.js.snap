exports[`components <ActivityListItem /> renders with hyperlinks 1`] = `
<div
  className="m-b-3">
  <div
    className="row">
    <div
      className="col-xs-12">
      <p
        className="m-b">
        <span
          className="title-label">
          Test 24
        </span>
        <span
          className="subtitle-label">
           - 
          Fri Dec 02 2016 (Ride)
        </span>
      </p>
      <table>
        <tbody>
          <tr>
            <td
              width="0">
              <span
                className="m-r">
                <em>
                  Distance
                </em>
              </span>
            </td>
            <td
              width="0">
              <span
                className="m-r">
                0
                 km
              </span>
            </td>
            <td
              width="0">
              <span
                className="m-r">
                <em>
                  Elapsed Time
                </em>
              </span>
            </td>
            <td
              width="0">
              <span
                className="m-r">
                5
                 minutes
              </span>
            </td>
            <td
              className="m-r-2">
              <span
                className="m-r">
                <em>
                  Type
                </em>
              </span>
            </td>
            <td>
              Ride
            </td>
          </tr>
          <tr>
            <td
              width="0">
              <span
                className="m-r">
                <em>
                  Avg Speed
                </em>
              </span>
            </td>
            <td
              width="0">
              <span
                className="m-r">
                NaN
                 kph
              </span>
            </td>
            <td
              width="0">
              <span
                className="m-r">
                <em>
                  Max Speed
                </em>
              </span>
            </td>
            <td>
              34
               kph
            </td>
            <td
              className="m-r-2">
              <span
                className="m-r">
                <em>
                  Trainer
                </em>
              </span>
            </td>
            <td>
              Yes
            </td>
          </tr>
        </tbody>
      </table>
      <div
        className="m-t">
        <a
          className="m-r-2"
          onClick={[Function]}
          style={Object {}}>
          Compare Segments
        </a>
        <a
          className="m-r-2"
          onClick={[Function]}
          style={Object {}}>
          3D Route Profile
        </a>
      </div>
    </div>
  </div>
  <hr />
</div>
`;

exports[`components <ActivityListItem /> renders without hyperlinks when there are no segments 1`] = `
<div
  className="m-b-3">
  <div
    className="row">
    <div
      className="col-xs-12">
      <p
        className="m-b">
        <span
          className="title-label">
          Test 24
        </span>
        <span
          className="subtitle-label">
           - 
          Fri Dec 02 2016 (Ride)
        </span>
      </p>
      <table>
        <tbody>
          <tr>
            <td
              width="0">
              <span
                className="m-r">
                <em>
                  Distance
                </em>
              </span>
            </td>
            <td
              width="0">
              <span
                className="m-r">
                0
                 km
              </span>
            </td>
            <td
              width="0">
              <span
                className="m-r">
                <em>
                  Elapsed Time
                </em>
              </span>
            </td>
            <td
              width="0">
              <span
                className="m-r">
                5
                 minutes
              </span>
            </td>
            <td
              className="m-r-2">
              <span
                className="m-r">
                <em>
                  Type
                </em>
              </span>
            </td>
            <td>
              Ride
            </td>
          </tr>
          <tr>
            <td
              width="0">
              <span
                className="m-r">
                <em>
                  Avg Speed
                </em>
              </span>
            </td>
            <td
              width="0">
              <span
                className="m-r">
                NaN
                 kph
              </span>
            </td>
            <td
              width="0">
              <span
                className="m-r">
                <em>
                  Max Speed
                </em>
              </span>
            </td>
            <td>
              34
               kph
            </td>
            <td
              className="m-r-2">
              <span
                className="m-r">
                <em>
                  Trainer
                </em>
              </span>
            </td>
            <td>
              Yes
            </td>
          </tr>
        </tbody>
      </table>
      <div
        className="m-t" />
    </div>
  </div>
  <hr />
</div>
`;

exports[`components <ActivityListItem /> renders without hyperlinks when there are segments but includeHyperlinks is false 1`] = `
<div
  className="m-b-3">
  <div
    className="row">
    <div
      className="col-xs-12">
      <p
        className="m-b">
        <span
          className="title-label">
          Test 24
        </span>
        <span
          className="subtitle-label">
           - 
          Fri Dec 02 2016 (Ride)
        </span>
      </p>
      <table>
        <tbody>
          <tr>
            <td
              width="0">
              <span
                className="m-r">
                <em>
                  Distance
                </em>
              </span>
            </td>
            <td
              width="0">
              <span
                className="m-r">
                0
                 km
              </span>
            </td>
            <td
              width="0">
              <span
                className="m-r">
                <em>
                  Elapsed Time
                </em>
              </span>
            </td>
            <td
              width="0">
              <span
                className="m-r">
                5
                 minutes
              </span>
            </td>
            <td
              className="m-r-2">
              <span
                className="m-r">
                <em>
                  Type
                </em>
              </span>
            </td>
            <td>
              Ride
            </td>
          </tr>
          <tr>
            <td
              width="0">
              <span
                className="m-r">
                <em>
                  Avg Speed
                </em>
              </span>
            </td>
            <td
              width="0">
              <span
                className="m-r">
                NaN
                 kph
              </span>
            </td>
            <td
              width="0">
              <span
                className="m-r">
                <em>
                  Max Speed
                </em>
              </span>
            </td>
            <td>
              34
               kph
            </td>
            <td
              className="m-r-2">
              <span
                className="m-r">
                <em>
                  Trainer
                </em>
              </span>
            </td>
            <td>
              Yes
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
  <hr />
</div>
`;
